uarray/_uarray_dispatch.cxx

```diff
@@ -309,14 +310,21 @@ class context_helper
 
   bool exit()
     {
-      bool success = (enter_size_ + 1 == backends_->size()
-                      && backends_->back() == new_backend_);
-      if (enter_size_ < backends_->size())
-        backends_->resize(enter_size_);
-
-      if (!success)
+      bool success = true;
+      if (sizes_.size() == 0 || backends_->size() != sizes_.back()) {
```

Comment:
If `sizes_.size() == 0`then calling `pop_back` is undefined behaviour.

Reply:
What do you suggest we do in this case? I'm inclined to raise a `SystemError`.

Reply:
What was wrong with the way it was done before?
```
      if (enter_size_ < backends_->size())
        backends_->resize(enter_size_);

```

Reply:
`enter_size_` may not be defined if `enter_` was not called.

Reply:
Ah of course, `sizes_.size()` was making me think of no backends, rather than no matching `__enter__` call.